java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -761,6 +812,16 @@ public Builder withCallOptions(final Collection<CallOption> options) {
       return this;
     }
 
+    /**
+     * Sets the catalog for this handler.
+     * @param catalog the catalog
+     * @return this instance.
+     */
+    public Builder withCatalog(final String catalog) {
+      this.catalog = catalog;
```

Comment:
Why not store Optional<String> in the first place?

Reply:
Hmm, I made Optional now, which means I pass Optional as a parameter into the constructor.  Not feeling that is the best practice, but have restricted the scope of createNewHandler() to package. Added extra unit tests to safe guard against possible NPE later.

Reply:
Ah...what you had originally was fine then, but I was more questioning if empty string for catalog is valid (can't a system have an unnamed default catalog?) in which case it would be better to only differentiate between null/not-null. I proposed Optional to be explicit about the semantics but certainly `@Nullable String` is fine

Reply:
Haha, yes. Hopefully the accepted values are clear now, but Optional might be overkill.
Yes a system having an unnamed default catalog is a reasonable use-case.

Reply:
Ah ok so then maybe this bit should be adjusted?

```
this.catalog = Optional.ofNullable(catalog).map(String::trim).filter(str -> !str.isEmpty());
```

Reply:
Thinking about this again. If the client has set a catalog value that is not null then we should pass this through for validation by the server. The client shouldn't be doing that. 
If the catalog is null then that should not be passed to the server. If the server supports a default, this is the case when that will be used.